test/integration/add-referral-email-controller.spec.ts

```diff
@@ -0,0 +1,220 @@
+import { test } from '../components'
+import { createTestIdentity, Identity, makeAuthenticatedRequest } from './utils/auth'
+import { TestCleanup } from '../db-cleanup'
+
+test('POST /v1/referral-email', function ({ components }) {
+  let cleanup: TestCleanup
+
+  beforeAll(async () => {
+    cleanup = new TestCleanup(components.pg)
+  })
+
+  describe('when setting a referral email', () => {
+    let identity: Identity
+    let userAddress: string
+
+    beforeEach(async () => {
+      identity = await createTestIdentity()
+      userAddress = identity.realAccount.address.toLowerCase()
+    })
+
+    afterEach(async () => {
+      await cleanup.cleanup()
+    })
+
+    describe('and the request is not signed', () => {
+      let validEmail: string
+
+      beforeEach(() => {
+        validEmail = 'test@example.com'
+      })
+
+      it('should respond with a 400 status code', async () => {
+        const { localHttpFetch } = components
+        const response = await localHttpFetch.fetch('/v1/referral-email', {
+          method: 'POST',
+          body: JSON.stringify({ email: validEmail })
+        })
+        expect(response.status).toBe(400)
+      })
+    })
+
+    describe('and the request is signed', () => {
+      describe('with valid email', () => {
+        let validEmail: string
+
+        beforeEach(() => {
+          validEmail = 'test@example.com'
+        })
+
+        it('should set referral email and return 204', async () => {
+          const response = await makeAuthenticatedRequest(components)(identity, '/v1/referral-email', 'POST', {
+            email: validEmail
+          })
+          expect(response.status).toBe(204)
+        })
+
+        it('should set referral email with email containing spaces and return 204', async () => {
+          const response = await makeAuthenticatedRequest(components)(identity, '/v1/referral-email', 'POST', {
+            email: '  test@example.com  '
+          })
+          expect(response.status).toBe(204)
+        })
+      })
+
+      describe('with invalid email format', () => {
```

Comment:
WDYT about making all the `it` statements inside this `describe` a describe on their own? Like:

```
describe('with invalid email format'
  describe('when email is missing'
    it('should respond with a 400 status code'
```